### ISSUE TITLE
resourceOrder wrong when used with resourceGroupField 

### ISSUE THREAD

**Issue by StephenWRogers** [fullcalendar/fullcalendar#5482]
### Reproduction

https://codepen.io/stephenwrogers/pen/MWKYYzz?editors=0010

### Description

This is probably the issue with #5094 too - if you use resourceGroupField in combination with resourceOrder it doesn't work as expected.

I'd expect group 1 with order 1 to be before of group 2 with order 2 here - if you comment out resourceGroupField you see the right ordering (B1, B2, A1, A2), but with the grouping the ordering is wrong.

**Comment by acerix** [comment]
I'm not sure it's a bug, the ordering works within each group and the order of the groups can be set by ordering first by the group:

https://codepen.io/acerix/pen/oNbgOdK?editors=0010

**Comment by StephenWRogers** [comment]
It only works if the field you specify in the ordering is the same as the one in resourcegroupfield. If you add an identical property called "grouporder" with the same values as group and order by "grouporder,order" it doesn't work.

The problem I guess is that the field I want to use for the grouping isn't the same one as for ordering.

**Comment by StephenWRogers** [comment]
See https://codepen.io/stephenwrogers/pen/Bajyeow?editors=0010 for last comment.

It looks like this only orders the groups if one of the ordering fields matches the one used for grouping (hence it works for "group,order" and not "grouporder,order")?

https://github.com/fullcalendar/fullcalendar-scheduler/blob/9f604256aca75b7410697829e3d33e05b371c35b/resource-timeline/src/ResourceTimelineView.tsx#L439

**Comment by acerix** [comment]
Yes, I couldn't find a built-in way to order the groups like that, but since they appear in the order they are defined, you could sort your resources by the "grouporder" before adding to the calendar as a workaround.

**Comment by StephenWRogers** [comment]
That's what I'd tried to do initially, I'd ordered the resources in the order I wanted them to appear on the calendar and not defined any resourceOrder, but they don't appear in the same order as defined.

I've managed to reproduce that now - https://codepen.io/stephenwrogers/pen/rNxOamY?editors=0010 - it looks like if no resourceOrder is defined the resources are ordered by the "id" property, rather than the order they're defined.

Which means I can't see a way to order the way I want 

- if I don't define a resourceOrder they order by "id" - which is not necessarily the right order as it's the ID from the database
- if i do define a resourceOrder it has to be the same as the resourceGroupField (team) which doesn't work either.


My only workaround, I think, is to output the database ID of the resource in a different field, and then manually generate the "id"s based on the order I want, but that doesn't seem right at all, or the expected behaviour?

**Comment by StephenWRogers** [comment]
Oh I see this is also here  https://github.com/fullcalendar/fullcalendar/issues/5301 

If I didn't have the grouping I could output an order field and order by that, but with the grouping (on a field I don't want to sort by) I can't see an easy way of rendering in the way I want them to show.

**Comment by acerix** [comment]
Since the bug only affects numeric id's, you could prefix the id's with a space or similar to make them "non-numeric" which is not the nicest hack but seems to work ok:

https://codepen.io/acerix/pen/bGdLvmM?editors=0010

**Comment by Fixoon** [comment]
Is there any update on this issue? I have the same issue. I want the order of the resource groups to be defined by something else than the resourceGroupField or resource Id. Preferably the order property of the resource. Or some separate field like "resourceGroupOrder".

**Comment by fabianosanttana** [comment]
Any update on this issue?

**Comment by AlaDouagi** [comment]
I managed to order resources **by the order of the array passed** just by overriding the default sorting parameter with `null`.

**React example**:
```jsx
  <FullCalendar
    initialView="resourceTimelineMonth"
    headerToolbar={{
      left: 'prev,today,next',
      center: 'title',
      right: null,
    }}
    plugins={[resourceTimelinePlugin]}
    resources={resources}
    events={events}
    // Sorting by index with the option below 👇
    resourceOrder={null}
    {...props}
  />
```

**Comment by akaspick** [comment]
@AlaDouagi Nice. null works. This really should sort integers properly otherwise in my case I had 9 come after 10 when "sorted".

**Comment by zwarrell** [comment]
Any update on this issue?  I am having the same problem.  Ideally would could sort by something other than the resoucrce group id itself!  See this code pen as an example.  Any advice is appreciated.

https://codepen.io/zwarrell/pen/RwmbOBy?editors=0010

**Comment by LucasBerbesson** [comment]
Same here, adding then removing a number in front of the resource group name : 

```
resourceGroupLabelContent: function (arg) {
        return {'html': `${arg.groupValue.slice(1)}`}
}
```

**Comment by enisyugnak** [comment]
I think this is still unsolved, and thanks for your approach I'm able to sort correctly. Just one minor addition to that solution for further reference if anybody needs. If there is more than 10 resource group above solution will not work, I mean it will work but include the number if its greater than 9 so I have changed it a little bit like this, 

I put - between number and string  so it becomes like this,  1-item 2-item .... 12-item
then I wipe them out... 

`resourceGroupLabelContent: function (arg) {
  const value = arg.groupValue;
  const title = value.includes("-") ? value.split("-")[1] : str;
        return {'html': `${title}`}
}`

hope it helps.

**Comment by arshaw** [comment]
It looks like the most recent version of FullCalendar (v6.1.18) does not allow the `resourceOrder={null}` HACK. See: https://codepen.io/arshaw/pen/EaVvvgg?editors=0010

The best way around this is to have an `index` property, like this: https://codepen.io/arshaw/pen/JoYyyRW?editors=0010

I'll reassess for a future release.
